src/main/kotlin/com/petqua/application/order/OrderService.kt

```diff
@@ -115,22 +115,87 @@ class OrderService(
                 orderName = orderName,
                 orderShippingAddress = OrderShippingAddress.from(shippingAddress, command.shippingRequest),
                 orderProduct = productCommand.toOrderProduct(
-                    shippingNumber = ShippingNumber.of(product.storeId, productCommand.deliveryMethod, orderNumber),
-                    product = product,
-                    storeName = storesById[product.storeId]?.name ?: throw OrderException(PRODUCT_NOT_FOUND),
+                    shippingNumber = ShippingNumber.of(
+                        productSnapshot.storeId,
+                        productCommand.deliveryMethod,
+                        orderNumber
+                    ),
+                    productSnapshot = productSnapshot,
+                    storeName = storesById[productSnapshot.storeId]?.name ?: throw OrderException(PRODUCT_NOT_FOUND),
                 ),
                 isAbleToCancel = true,
                 status = ORDER_CREATED,
                 totalAmount = command.totalAmount,
             )
         }
-        orderRepository.saveAll(orders)
+        return orderRepository.saveAll(orders)
+    }
+}
+
+class OrderProducts(
+    val productById: Map<Long, Product>,
+    val productOptions: Set<ProductOption>,
+) {
+    val productIds = productById.keys.toList()
+    val products = productById.values.toList()
```

Comment:
constructor로 위에 필드랑 일관되게 관리할 수 있을 것 같아요!
```suggestion
class OrderProducts(
    val productById: Map<Long, Product>,
    val productOptions: Set<ProductOption>,
    val productIds: List<Long>,
    val products: List<Product>,
) {

    constructor(productById: Map<Long, Product>, productOptions: Set<ProductOption>) : this(
        productById,
        productOptions,
        productById.keys.toList(),
        productById.values.toList()
    )
```

Reply:
오 훨씬 낫네요 반영했습니다!👍